docs/enhancements/cluster_pool_hot_spares.md

```diff
@@ -0,0 +1,111 @@
+
+### Non-Goals
+
+Solving world peace.
+
+
+### User Story
+
+As a consumer of a (properly tuned) ClusterPool,
+
+I want ClusterClaim to give me an active cluster immediately in the common case,
+
+So that I can get my work done more quickly.
+
+(Ideally without sacrificing the cost savings of having some clusters in the pool hibernating.)
+
+## Proposals
+
+Several implementation alternatives have already come to mind.
+These are not necessarily mutually exclusive.
+
+### ClusterClaim Prioritizes Active Clusters
+
+For most/all of the options below, it makes sense to add logic to ClusterClaim to preferentially grab an Active cluster first.
+The simplest implementation is to sort the list of ClusterDeployments into three buckets: Active, Resuming, and Hiberating, claimed in that order.
+A smarter implementation could take into account when Resuming clusters were asked to resume, with the most recent at the end, on the theory that the least-recently-resumed will be ready first.
+
+### ClusterClaim Must Account For `hibernateAfter`
+
+This applies to most/all of the options below.
+
+If the ClusterPool has `hibernateAfter` set, that clock starts running as soon as the ClusterDeployment is ready.
+(TODO: Confirm this -- does the clock start at creation, or `installed == True`, or something else?)
+If we don't handle this properly for Active pool members, your cluster might go to sleep unexpectedly while you're working (Narcoleptic Cluster Syndrome).
+
+**Options:**
+- Reset the `hibernateAfter` clock at claim time. (How? Is this something hive controls?)
+- Make `hibernateAfter` mutually exclusive with whatever solution(s) we pick below.
+  (I think we would have to do this with a validation webhook.)
+
+### Don't Hibernate New ClusterDeployments
+
+Today when ClusterDeployments are created to fill out the pool, they are created with `PowerState = HibernatingClusterPowerState`.
+Add an option to ClusterPool to not do that.
+
+**Pros:**
+- Dead simple
+
+**Cons:**
+- In periods of low activity, you're burning money by having Active vs. Hibernating clusters.
+  (I suppose you could manually hibernate clusters in the pool. Not the best UX.)
+
+### `activeCount`/`activePercent`
+
+Add knob(s) to ClusterPool indicating a portion of the pool to maintain in Active state.
+Either or both (if both, they should be mutually exclusive, or one should take precedence) of:
+
+- `activeCount`: Integer absolute number of Active clusters to strive for.
+  Constrained to `0 (default) <= activeCount <= (Max?)Size`.
+- `activePercent`: Float `0.0 (default) <= activePercent <= 100.0` percentage of `(Max?)Size` clusters to keep Active.
+  (How should we round this?)
+
+Either way, this translates to an integer number of clusters the ClusterPool controller will try to keep Active.
+- When creating a new cluster for the pool, set `PowerState = HibernatingClusterPowerState` iff we're at/above that threshold.
```

Comment:
We need to be careful about this algorithm so we don't end up LIFOing CDs. Viz:
- Pool size is 3, activeCount is 1. We set up 2 hibernating and 1 active
- Claim 1. You get the active one.
- We spin up a new CD to bring the count back to 3. We check against active count to figure out whether to hibernate the new one or not. We decide not.

So the newest one is the active one. Repeat. The two hibernating ones never get used.

I think to mitigate this, we need to fully decouple hibernation state from new CD creation. In the reconcile loop, *after* we've done assignments and kicked off any installations, we calculate how many clusters need to be hibernated or resumed. We decide which ones based on overall age (`creationTimestamp`), which means we'll prioritize hibernating Installing clusters and resuming the oldest Hibernating clusters.

Reply:
Updated doc accordingly.